scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/solver/PrioritySolver.java

```diff
@@ -666,11 +679,43 @@ else if(!analysisOnly &&  (missing instanceof MissingActivityTemplateConflict mi
             narrowed = narrowByResourceConstraints(actWindow, List.of(stateConstraints));
           }
           if(narrowed.includes(actWindow)){
-            //decision-making here, we choose the first satisfying activity
-            evaluation.forGoal(goal).associate(act, false);
-            itConflicts.remove();
-            logger.info("Activity " + act + " has been associated to goal " + goal.getName() +" to satisfy conflict " + i);
-            break;
+            // If existing activity is a match but is missing the anchor, then the appropriate anchorId has been included in MissingAssociationConflict.
+            // In that case, a new activity must be created as a copy of act but including the anchorId. This activity is then added to all appropriate data structures and the association is created
+            if (missingAssociationConflict.getAnchorIdTo().isPresent()){
+              SchedulingActivityDirective predecessor = plan.getActivitiesById().get(missingAssociationConflict.getAnchorIdTo().get());
+              Duration startOffset = act.startOffset().minus(plan.calculateAbsoluteStartOffsetAnchoredActivity(predecessor));
+              // In case the goal requires generation of anchors, then check that the anchor is to the Start. Otherwise (anchor to End), make sure that there is a positive offset
+              if(missingAssociationConflict.getAnchorToStart().isEmpty() || missingAssociationConflict.getAnchorToStart().get() || startOffset.longerThan(Duration.ZERO)){
+                var replacementAct = SchedulingActivityDirective.copyOf(
+                    act,
+                    missingAssociationConflict.getAnchorIdTo().get(),
+                    missingAssociationConflict.getAnchorToStart().get(),
+                    startOffset
+                );
+                replaceActivity(act,replacementAct);
+                act = replacementAct;
+                //decision-making here, we choose the first satisfying activity
```

Comment:
why replace `act` by `replacementAct` ? Can't you just use `replacementAct` in the statement below ? as it is the iterator variable for values in actToChooseFrom, I find it odd to assign it.

Reply:
agree. removed assignment. replacementAct used directly